exp/operator/controllers/shared.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/cluster-api/api/v1alpha4"
+	operatorv1 "sigs.k8s.io/cluster-api/exp/operator/api/v1alpha4"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func preflightChecks(ctx context.Context, c client.Client, gvk schema.GroupVersionKind) (*v1alpha4.Condition, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	log.V(4).Info("Performing flight checks.")
+
+	providerList := &unstructured.UnstructuredList{}
+	providerList.SetGroupVersionKind(gvk)
+
+	if err := c.List(ctx, providerList); err != nil {
+		return nil, errors.Wrap(err, "failed to list providers")
+	}
+
+	// Checking if more than one instance of the same provider exists in the cluster.
+	if len(providerList.Items) > 1 {
+		log.V(4).Info("More than one core provider exists in the cluster. Only one is allowed.")
+		return &v1alpha4.Condition{
```

Comment:
Please use `conditions.FalseCondition` to ensure consistency across the codebase